app/controllers/public/orders_controller.rb

```diff
@@ -52,14 +57,22 @@ def confirm
       @order.address = current_customer.address
       @order.address_name = current_customer.first_name + current_customer.last_name
     elsif params[:order][:address_option] == "1"
-      @address = Address.find(params[:customer_address])
-      @order.postal_code = @address.postal_code
-      @order.address = @address.address
-      @order.address_name = @address.address_name
+      if params[:customer_address].present?
+        @address = Address.find(params[:customer_address])
+        @order.postal_code = @address.postal_code
+        @order.address = @address.address
+        @order.address_name = @address.address_name
+      else
+        redirect_to new_order_path, alert: "登録済住所がありません"
+      end
```

Comment:
最後にバリデーションチェックを行ってリダイレクト先を決めるので、ここでの指定は不要です。
```suggestion
      if params[:customer_address].present?
        @address = Address.find(params[:customer_address])
        @order.postal_code = @address.postal_code
        @order.address = @address.address
        @order.address_name = @address.address_name
      end
```